app/src/main/java/com/poupa/vinylmusicplayer/misc/queue/StaticPlayingQueue.java

```diff
@@ -0,0 +1,458 @@
+        return queue;
+    }
+
+    public ArrayList<IndexedSong> getOriginalPlayingQueue() {
+        return originalQueue;
+    }
+
+    private void resetSongs() {
+        this.songs.clear();
+
+        for (IndexedSong song : queue) {
+            songs.add((Song)song);
+        }
+        songsIsStale = false;
+    }
+
+    public ArrayList<Song> getPlayingQueueSongOnly() {
+        if (songsIsStale)
+            resetSongs();
+        return songs;
+    }
+
+    public int size() {
+        return queue.size();
+    }
+
+    /* -------------------- position method -------------------- */
+
+    public int getCurrentPosition() {
+        return currentPosition;
+    }
+
+    public void setCurrentPosition(int position) {
+        if (position >= queue.size())
+            return;
+
+        currentPosition = position;
+    }
+
+    public void setPositionToNextPosition() {
+        currentPosition = nextPosition;
+    }
+
+    public void setNextPosition(int position) {
+        if (position >= queue.size())
+            return;
+
+        nextPosition = position;
+    }
+
+    public boolean isLastTrack() {
+        return getCurrentPosition() == queue.size() - 1;
+    }
+
+    public int getNextPosition(boolean force) {
+        int position = getCurrentPosition() + 1;
+        switch (getRepeatMode()) {
+            case REPEAT_MODE_ALL:
+                if (isLastTrack()) {
+                    position = 0;
+                }
+                break;
+            case REPEAT_MODE_THIS:
+                if (force) {
+                    if (isLastTrack()) {
+                        position = 0;
+                    }
+                } else {
+                    position -= 1;
+                }
+                break;
+            default:
+            case REPEAT_MODE_NONE:
+                if (isLastTrack()) {
+                    position -= 1;
+                }
+                break;
+        }
+        return position;
+    }
+
+    public int getPreviousPosition(boolean force) {
+        int newPosition = getCurrentPosition() - 1;
+        switch (repeatMode) {
+            case REPEAT_MODE_ALL:
+                if (newPosition < 0) {
+                    newPosition = queue.size() - 1;
+                }
+                break;
+            case REPEAT_MODE_THIS:
+                if (force) {
+                    if (newPosition < 0) {
+                        newPosition = queue.size() - 1;
+                    }
+                } else {
+                    newPosition = getCurrentPosition();
+                }
+                break;
+            default:
+            case REPEAT_MODE_NONE:
+                if (newPosition < 0) {
+                    newPosition = 0;
+                }
+                break;
+        }
+        return newPosition;
+    }
+
+    /* -------------------- song getter info -------------------- */
+
+    public long getQueueDurationMillis(int position){
+        long duration = 0;
```

Comment:
is it intentional that the queue duration is from a position and not for the whole queue?